### PR TITLE
Fix issue 12 and extend CI tests to additional Python versions/OSes

### LINKED ISSUE #12: build.py errors for Python 3.6 on Windows
This error came up by running the tests from this repo as part of the IDAES/idaes-pse integration tests. Here's the error log:

```txt
============================= test session starts =============================
platform win32 -- Python 3.6.8, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: D:\a\idaes-pse\idaes-pse\examples-pse, configfile: pytest.ini
plugins: cov-2.10.1
collected 1 item / 2 errors

=================================== ERRORS ====================================
_________________ ERROR collecting tests/test_build_script.py _________________
tests\test_build_script.py:18: in <module>
    import build
build.py:120: in <module>
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
E   AttributeError: module 'asyncio' has no attribute 'WindowsSelectorEventLoopPolicy'
------------------------------- Captured stdout -------------------------------
path: ['D:\\a\\idaes-pse\\idaes-pse\\examples-pse', 'D:\\a\\idaes-pse\\idaes-pse\\examples-pse\\tests', 'C:\\hostedtoolcache\\windows\\Python\\3.6.8\\x64\\Scripts\\pytest.exe', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\python36.zip', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\DLLs', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\win32', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\win32\\lib', 'c:\\hostedtoolcache\\windows\\python\\3.6.8\\x64\\lib\\site-packages\\Pythonwin']
__________________ ERROR collecting tests/test_notebooks.py ___________________
tests\test_notebooks.py:17: in <module>
    import build
build.py:120: in <module>
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
E   AttributeError: module 'asyncio' has no attribute 'WindowsSelectorEventLoopPolicy'
```

. It seems to be limited to the specific combination Python 3.6 and Windows, and wasn't caught by the tests here in examples-pse because the CI is not running on the full Python version and OS test matrix just yet (which will be rectified ASAP).

I think the fix shouldn't be too hard, essentially just add a special case within the special case in build.py#L117-120.

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## Proposed changes:
- Add an explicit Python version check to exclude incompatible versions from workaround
- Extend CI tests so that they run on Python version/OS combination matrix

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -114,7 +114,8 @@
 
 
 # This is a workaround for a bug in some versions of Tornado on Windows for Python 3.8
-if sys.platform == "win32":
+# the sys.version_info check includes the micro version, so e.g. 3.7.1 will also evaluate to True
+if sys.platform == "win32" and sys.version_info > (3, 7):
```
Comment (dangunter) [inline build.py:118]: Do we know that this is not fixed in 3.9? I prefer our hack have as short a lifetime as possible

**Review by dangunter — DISMISSED** [review]
Looks good

**Review by andrewlee94 — APPROVED** [review]
Rubber stamp.

**Comment by lbianchi-lbl** [comment]
@dangunter not sure what's the status for Python 3.9 but we can try to see how it behaves if we remove the workaround once we start supporting 3.9.

In the meanwhile, what I'm getting from tornadoweb/tornado#2608 is that, for the time being, this should be dealt with on the side of the application: so our workaround is the recommended way to address this. In any case he issue looks quite active and we can use as a reference to see if there's any progress.

I've added this information as a comment in the code, so if you're OK with the wording you can go ahead and re-approve it.

**Review by andrewlee94 — APPROVED** [review]
Looks good to me.

**Review by ksbeattie — APPROVED** [review]
LGTM

**Pull request merged**